Training/variable_size_arrays.cpp

```diff
@@ -1,21 +1,24 @@
 #define _CRT_SECURE_NO_WARNINGS
 #include <cstdio>
 #include <vector>
+
 #include "variable_size_arrays.h"
 
 int hackerrank::variable_size_arrays()
 {
     // parse line 1 -- n (number of arrays), q (number of queries)
-    int n, q;
+    int n = -1, q = -1; //ES- Habitually instantiate variables with invalid defaults.
     scanf("%d %d", &n, &q);
     
-    // parse arrays
+    // parse arrays - ES- consider helper functions even for problems like these.
     std::vector<std::vector<int>> matrix;
-    for (int i = 0; i < n; i++) {
+    matrix.reserve(n); //ES- nit: we can frontload the resizing costs since we know the size.
+    for (int i = 0; i < n; ++i) { // ES- nit: prefix incrementation is preferred over postfix inc
```

Comment:
Pretty much always preferred when it's a standalone statement. A compiler will optimize appropriately in this case, but if the type of `i` isn't an `int` and/or `++` was overloaded, the prefix increment operator is less expensive than the postfix operator (due to the creation of a temp object).

It doesn't really matter too much with primitives, but if you ever convert code from using primitives into a `template`, it can save some time to have already used prefix operators.

It's one of those things that doesn't _really_ matter but because it _could_ become a template in the future we'd just rather people use prefix version.